consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -85,18 +86,24 @@ trait BaseApi extends Closable {
   mapper.registerModule(DefaultScalaModule)
   mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
-  def parseJson[T: Manifest](buf: Buf): Try[T] = {
-    val Buf.ByteArray.Owned(bytes, begin, end) = Buf.ByteArray.coerce(buf)
-    Try(mapper.readValue[T](bytes, begin, end - begin))
-  }
+  private def parse[T: Manifest](rsp: Response): Future[T] =
+    for {
+      data <- rsp match {
+        case chunked if rsp.isChunked => chunked.chunkReader.accumulate.map({ case (buf, _) => Buf.ByteArray.coerce(buf) })
```

Comment:
We don't need access to the trailers (indeed, we discard them here) so we can use the slightly more ergonomic `.reader` API instead of `.chunkReader`.  

I think this could simply be `Reader.readAll(chunked.reader)`